pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -60,6 +60,12 @@ func (r *pullthroughBlobStore) remoteStat(ctx context.Context, dgst digest.Diges
 		return distribution.Descriptor{}, err
 	}
 
+	r.pullFromInsecureRegistries = false
```

Comment:
I agree with @smarterclayton and @liggitt and I think that this setting should be per imagestream. In this case we should ignore global value to avoid side effects.